### PR TITLE
Add presubmit suite with etcd3 underneath

### PR DESCRIPTION
**Pull request by wojtek-t**
@fejta @ixdy The project doesn't have enough quota yet, but for now I'm only enabling manual tests of it.
Once the project will be fully ready and I will be sure about it, I will send a separate PR to run this presubmit for all PRs.

Ref https://github.com/kubernetes/kubernetes/issues/20504
@mml @lavalamp - what we discussed.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/test-infra/779)

### CODE REVIEW

**Comment by wojtek-t** [comment]
@fejta @ixdy - this failure looks at least strange to me:
```
--- FAIL: TestJobTriggers (0.00s)
	jobs_test.go:48: For job kubernetes-pull-build-test-e2e-gce-etcd3: RerunCommand "@k8s-bot (gce )?etcd3 (e2e )?test this" does not match regex "@k8s-bot (gce )?etcd3 (e2e )?test this".
```

For me those are equal strings...

**Comment by spxtr** [comment]
The rerun command is what people should say to rerun it. For this case, `@k8s-bot gce etcd3 e2e test this` would make sense. I should probably clarify that in the documentation.

**Comment by wojtek-t** [comment]
@spxtr - the documentation (and the meaning of it is pretty clear). I'm just asking why the test failed, since I don't understand it...

**Comment by spxtr** [comment]
The test makes sure that the rerun command is matched by the trigger string. The string `@k8s-bot (gce )?etcd3 (e2e )?test this` is not matched by the regex `@k8s-bot (gce )?etcd3 (e2e )?test this`.

**Comment by wojtek-t** [comment]
@spxtr - those two strings are equal. How it is not matched?

**Comment by wojtek-t** [comment]
@fejta explained it to me offline. Should be fixed now.

**Comment by wojtek-t** [comment]
@fejta @ixdy - can you please take a look?

File: jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull-e2e.yaml [hunk 1]
```diff
@@ -141,6 +141,63 @@
                   echo "Exiting with code: ${{rc}}"
                   exit ${{rc}}
                 fi
+        - 'e2e-gce-etcd3': # kubernetes-pull-build-test-e2e-gce-etcd3
+            cmd: |
+                export KUBE_SKIP_PUSH_GCS=y
+                export KUBE_RUN_FROM_OUTPUT=y
+                export KUBE_FASTBUILD=true
+                ./hack/jenkins/build.sh
+                if [[ "${{ghprbTargetBranch:-}}" == "release-1.0" || "${{ghprbTargetBranch:-}}" == "release-1.1" ]]; then
```
Comment (ixdy) [inline jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull-e2e.yaml]: you probably don't need this.

**Comment by wojtek-t** [comment]
@spxtr @ixdy - can you please take a look?

**Comment by fejta** [comment]
LGTM to unblock things in case Joe and Jeff don't have time to review given the time difference.

**Comment by spxtr** [comment]
LGTM

**Comment by wojtek-t** [comment]
Thanks!

**Pull request merged**